### PR TITLE
Added Tutorial Street Scene

### PR DESCRIPTION
**Pull request by Amberg83**
The Tutorial Street has spaced out enemies
The area was fully rebuilt to accomodate for potential future automatic map generation

### CODE REVIEW

File: Assets/!/Scripts/PatrollingScript.cs [hunk 1]
```diff
@@ -8,7 +8,8 @@ public class PatrollingScript : MonoBehaviour
     public EnemyControllerScript enemyControllerScript;
     public string pathName;
     public bool isPatrolling = true;
-    
+    private Vector3 StartingPosition;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PatrollingScript.cs:11]:
```suggestion
    private Vector3 _startingPosition;
```

File: Assets/!/Scripts/PatrollingScript.cs [hunk 2]
```diff
@@ -49,4 +51,4 @@ private IEnumerator Wait(float waitTime)
         yield return new WaitForSeconds(waitTime);
         _isWaiting = false;
     }
-}
+}
```
Comment (matias-gonz) [inline Assets/!/Scripts/PatrollingScript.cs:54]: Missing line break

File: Assets/!/Scripts/TriggerPatrollingScript.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    void Start()
+    {
+        playerColliders = player.GetComponents<Collider>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
```
Comment (matias-gonz) [inline Assets/!/Scripts/TriggerPatrollingScript.cs:20]:
```suggestion
```

File: Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            obstaclesList.Append(segment.transform.Find("Obstacles").gameObject);
+            streetList.Append(segment.transform.Find("Street").gameObject);
+        }
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
```
Comment (matias-gonz) [inline Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs:30]:
```suggestion
```

File: Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class MapLoadingScript : MonoBehaviour
+{
+    [SerializeField] GameObject[] mapSegments;
+    private GameObject[] ambientList;
```
Comment (matias-gonz) [inline Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs:10]: Let's stick with C# conventions:
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names

```suggestion
    private GameObject[] _ambientList;
```

File: Assets/!/Scripts/TriggerPatrollingScript.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TriggerPatrollingScript : MonoBehaviour
```
Comment (matias-gonz) [inline Assets/!/Scripts/TriggerPatrollingScript.cs:5]: What is this script used for?
In what prefab is this added?

File: Assets/!/Scripts/PatrollingScript.cs [hunk 7]
```diff
@@ -39,7 +41,7 @@ void Update()
         else if (_hasReachedCurrentGoal)
         {
             _hasReachedCurrentGoal = false;
-            enemyControllerScript.MoveTo(_pathPoints[_currentPointIndex].Position, false);
+            enemyControllerScript.MoveTo(_pathPoints[_currentPointIndex].Position + StartingPosition, false);
```
Comment (Amberg83) [inline Assets/!/Scripts/PatrollingScript.cs:44]:
```suggestion
            enemyControllerScript.MoveTo(_pathPoints[_currentPointIndex].Position + _startingPosition, false);
```

File: Assets/!/Scripts/PatrollingScript.cs [hunk 8]
```diff
@@ -27,7 +29,7 @@ void Update()
             return;
         }
 
-        Vector3 position = transform.position;
+        Vector3 position = transform.position - StartingPosition;
```
Comment (Amberg83) [inline Assets/!/Scripts/PatrollingScript.cs:32]:
```suggestion
        Vector3 position = transform.position - _startingPosition;
```

File: Assets/!/Scripts/PatrollingScript.cs (hunk 2 above)
Comment (Amberg83) [reply Assets/!/Scripts/PatrollingScript.cs:54]: Why cant I resolve it from this panel like the other options?

File: Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs (hunk 5 above)
Comment (Amberg83) [reply Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs]: Is this correct now?

File: Assets/!/Scripts/PatrollingScript.cs (hunk 2 above)
Comment (Amberg83) [reply Assets/!/Scripts/PatrollingScript.cs]: Resolved it now

File: Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs (hunk 5 above)
Comment (Amberg83) [reply Assets/!/Prefabs/Level/Scripts/Map Loading Script.cs]: I think so xD

**Comment by Amberg83** [comment]
@matias-gonz I think I resolved everything but please take a look at it again if you have the time

File: .vsconfig
Comment (matias-gonz) [inline .vsconfig:1]: Shouldn't upload this file

File: Assets/!/Scripts/PatrollingScript.cs [hunk 9]
```diff
@@ -8,7 +8,8 @@ public class PatrollingScript : MonoBehaviour
     public EnemyControllerScript enemyControllerScript;
     public string pathName;
     public bool isPatrolling = true;
-    
+    private Vector3 _startingPosition;
```
Comment (matias-gonz) [inline Assets/!/Scripts/PatrollingScript.cs:11]: Very good improvement 👍

**Review by matias-gonz — CHANGES_REQUESTED** [review]

**Review by matias-gonz — APPROVED** [review]

**Pull request merged**